### PR TITLE
Put "Installed Plugins" & "Upload" plugin buttons behind feature gates

### PR DESCRIPTION
**Pull request by DustyReagan**
#### Proposed Changes

Redirects the following two pages to /plugins/{site_slug}

- /plugins/upload/{site_slug} [If missing `WPCOM_FEATURES_UPLOAD_PLUGINS`]
- /plugins/manage/{site_slug} [If missing `WPCOM_FEATURES_MANAGE_PLUGINS`]

Hides the "Installed Plugins" and "Upload" buttons if the site is missing the `WPCOM_FEATURES_MANAGE_PLUGINS` or `WPCOM_FEATURES_UPLOAD_PLUGINS` respectively.

Before | After
--|--
before-plugins | after-plugins

The "Upload" button on the /plugins/manage/{site_slug} is also gated by UPLOAD_PLUGINS. So if the site were to have MANAGE_PLUGINS but not UPLOAD_PLUGINS, the upload button will not appear.

Before | After
--|--
manage-before | manage-after

#### Testing Instructions

> **Note**
> Jetpack (non-Atomic) sites should always show the "Installed Plugins" & "Upload" plugin buttons, and the /plugins/upload/{site_slug} and /plugins/manage/{site_slug} pages should NOT redirect.

* Load this branch.
* Got to /plugins/{site_slug} on a sub Business/Pro plan.
  * You should NOT see the "Installed Plugins" or "Upload" buttons.
* Try manually going to /plugins/upload/{site_slug} & /plugins/manage/{site_slug}
  * You should be redirected to /plugins/{site_slug}
* Got to /plugins/{site_slug} on a Business/Pro or higher plan.
  * You should see the "Installed Plugins" and "Upload" buttons.
* Try manually going to /plugins/upload/{site_slug} & /plugins/manage/{site_slug}
  * You should NOT be redirected to /plugins/{site_slug}

This PR is related to #64116

### CODE REVIEW

File: client/my-sites/plugins/main.jsx [hunk 1]
```diff
@@ -37,6 +41,7 @@ import {
 	getSelectedSiteId,
 	getSelectedSiteSlug,
 } from 'calypso/state/ui/selectors';
+import siteHasFeature from '../../state/selectors/site-has-feature';
```
Comment (obenland) [inline client/my-sites/plugins/main.jsx]: It looks like the other imports from state use an absolute path. Let's update this and the others to `'calypso/state/selectors/site-has-feature'`?

**Review by obenland — COMMENTED** [review]
Nice work @DustyReagan!

**Comment by obenland** [comment]
I think we'll need to a check for Jetpack sites and only hide the buttons if it's a Simple or Atomic site. They're currently being hidden for all site types

**Comment by DustyReagan** [comment]
> I think we'll need to a check for Jetpack sites and only hide the buttons if it's a Simple or Atomic site. They're currently being hidden for all site types

Good catch! Thanks!

File: client/my-sites/plugins/main.jsx [hunk 2]
```diff
@@ -452,6 +467,12 @@ export default flow(
 			const visibleSiteIds = siteObjectsToSiteIds( getVisibleSites( sites ) ) ?? [];
 			const siteIds = siteObjectsToSiteIds( sites ) ?? [];
 			const pluginsWithUpdates = getPlugins( state, siteIds, 'updates' );
+			const jetpackNonAtomic =
+				isJetpackSite( state, selectedSiteId ) && ! isAtomicSite( state, selectedSiteId );
+			const hasManagePlugins =
+				siteHasFeature( state, selectedSiteId, WPCOM_FEATURES_MANAGE_PLUGINS ) || jetpackNonAtomic;
```
Comment (obenland) [inline client/my-sites/plugins/main.jsx:473]: Clever

**Review by obenland — APPROVED** [review]
:shipit:

**Pull request merged**